pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -70,7 +70,7 @@ func (uc *UpgradeCluster) UpgradeCluster(subscriptionID uuid.UUID, resourceGroup
 	}
 
 	switch ucs.OrchestratorProfile.OrchestratorVersion {
```

Comment:
Can you make sure upgrade works with your changes? E.g. OrchestratorVersion  was an input provided by the customer. Now OrchestratorVersionHint is the input, this is not the right field to be validated but it should flow into the upgrade operation to create the right template.